### ISSUE TITLE
Some potential questionable behaviour in the Installer

### ISSUE THREAD

**Issue by chrisbra** [git-for-windows/git#4216]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

this was Git-2.39.0.2-64-bit.exe

```
$ git --version --build-options

I don't have it here unfortunately
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

Version 11, but I am currently not running git-for-windows

```
$ cmd.exe /c ver

10.0.22621.1105
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

defaults

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
> type "$env:USERPROFILE\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

N/A
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

N/A I think

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

N/A

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

HI,
my company did run a security scan on the installer [Git-2.39.0.2-64-bit.exe](https://www.virustotal.com/gui/file/8cf0ee3efaabe8a9b9b6b6889ae0ed369d9f1c85696ad637e715959921ed71c3/detection). While virustotal looks okay, they noticed some potentialy unwanted behaviour and are a bit worried about some of those issues. 

https://otx.alienvault.com/indicator/file/8cf0ee3efaabe8a9b9b6b6889ae0ed369d9f1c85696ad637e715959921ed71c3

If you check the Alerts section, you'll notice 5 Alerts with High Severity. 2 of those (*infostealer_ftp*  and *detect_putty* ) I could explain with the putty integration in the [installer](https://github.com/git-for-windows/build-extra/blob/cfafe97ebde8095acef116488ff479175f25a40a/installer/install.iss#L2097-L2102), one other one (*Appends a known multi-family ransomware file extension to files that have been encrypted*) comes by distributing tcl encoding definition files (I assume this is for gui-gui in order to allow encoding of messages). 

That leaves: *Generates some ICMP traffic*  and *Communicates with host for which no DNS query was performed*. I cannot explain those. Would it be possible on why the installer would try to connect to some IPs directly without DNS querying and/or why it would try to generate some ICMP traffic?

Thanks, appreciate any respone here.

**Comment by dscho** [comment]
@chrisbra first of all, please review [Git for Windows' security policy](https://github.com/git-for-windows/git/security/policy): next time you report anything suspicious, please do so in a safer venue (i.e. the Git security mailing list).



> _Generates some ICMP traffic_ and _Communicates with host for which no DNS query was performed_. I cannot explain those. Would it be possible on why the installer would try to connect to some IPs directly without DNS querying and/or why it would try to generate some ICMP traffic?

I cannot explain those, either. Git for Windows does not do anything out of the ordinary, it's basically the InnoSetup installer doing its job, I think. But why it would require ICMP packets to be sent? No idea. The reverse lookup for the host it communicates with is a173-222-229-190.deploy.static.akamaitechnologies.com, which looks like it accesses _something_ via a CDN. That would potentially explain the lack of a DNS lookup: if you access resources via a Content Distribution Network, it may redirect to a host directly via numeric IP address. And that _may_ also explain the ICMP packets, essentially to figure out what is the closest host? But I do not know what it would download from a CDN, it's certainly not anything I would be aware of.

Would you have more details as to _what_ it tries to access, e.g. by using Fiddler while repeating the very same operation?

**Comment by chrisbra** [comment]
Thank you and I keep the security policy in mind for next time. I hope to find out some more from our sec guys.

**Comment by dscho** [comment]
> it's basically the InnoSetup installer doing its job, I think.

Indeed, innosetup's own installer also triggers the ICMP thing: https://otx.alienvault.com/indicator/file/50d21aab83579245f88e2632a61b943ad47557e42b0f02e6ce2afef4cdd8deb1

**Comment by dscho** [comment]
Since VirusTotal has no issue with the installer, I am chalking the time I spent on analyzing this up as yet another antivirus tax.

@chrisbra if you find out anything via Fiddler, please write down the information here.

**Comment by chrisbra** [comment]
Hey @dscho 
unfortunately, I haven't yet received more information here. But at least our security guys seem to be fine for now :) 

> Indeed, innosetup's own installer also triggers the ICMP thing: https://otx.alienvault.com/indicator/file/50d21aab83579245f88e2632a61b943ad47557e42b0f02e6ce2afef4cdd8deb1

Okay, but this only triggers the ICMP traffic,  not the  "Communicates with host for which no DNS query was performed".

So let me check with the Innosoft Forum (https://groups.google.com/g/innosetup/c/WIT4zamNkfg) what is going on there, although they are pretty strict about not asking questions for anti-virus program. Go figure...

Thanks appreciate your fast answer.

**Comment by dscho** [comment]
@chrisbra wouldn't it have made more sense to research a little e.g. using Fiddler as I suggested above, to figure out what is going on, and then accompany the bug report with said information? If it turns out to be a false alarm (it could be, for example, that something _else_ was responsible for the akamai CDN access, not the installer, and that the ICMP packets are sent by any Delphi program) and if that is the case, it would be potentially embarrassing not to have performed some preliminary analysis first.

**Comment by chrisbra** [comment]
Yes, agree. But it seems this is not possible at least for me unfortunately. I am just the messenger here.

**Comment by chrisbra** [comment]
So, I did spin up a local machine, installed Fiddler and try to find out something more. It's actually not clear to me, if Fiddler is able to capture ICMP traffic at all (Just for a test, I issued a ping and it didn't show up). Anyhow, I did run a git-for-windows installer using the brand-new Git-2.39.1-64-bit.exe and I just saw some traffic for OCSP Certificates of the smartscreen windows application, plus some Google Traffic (I believe this happened, when opening the Release Notes in Chrome browser). And nothing to that strange IP.

![grafik](https://user-images.githubusercontent.com/244927/213279302-90132871-cdab-4415-8972-6e6204c95910.png)

All in all, I would say it looks quite innocent.

**Comment by dscho** [comment]
> plus some Google Traffic (I believe this happened, when opening the Release Notes in Chrome browser)

It might have something to do with [Open Sans](https://github.com/git-for-windows/build-extra/blob/d714a6829b4068aff2f8e496a1405c75f0dfc949/ReleaseNotes.css#L2) being used

> It's actually not clear to me, if Fiddler is able to capture ICMP traffic at all

I don't believe it does. TBH I was interested exclusively in finding out what causes that akamai connection.

About the ICMP alert: I cannot replicate that. I fired up [Process Monitor](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon), started an installer, and then liberally excluded all calls, one by one, that could not have sent those pings. I ended up with an empty list. In other words, during _my_ experiment, no user-space programs issued ICMP requests.
